### DISCUSSION TITLE
Squash commit when merging PR

### DISCUSSION THREAD

**Discussion by caicancai** [tonbo-io/tonbo#108] [Ideas]
At present, I noticed that there was no squash commit when tonbo merged pr, which is actually not conducive to the learning of subsequent developers. I think this is an area that can be improved.

**Comment by caicancai** [comment]
what do you think? @ethe

**Comment by ethe** [comment]
Yes, Tonbo now chooses rebase and merge because we try to keep the atomicity of each previous commit. However, this is a very strict requirement, too many fragmented commits will make Tonbo history not easy to read. Squash and merge can make the history clean, but it will lose the details of each merge. Considering the visibility and readability of the balance details, is it a good idea for Tonbo to switch to normal merge?

**Reply by caicancai** [reply]
@ethe I didn't know the current merging method of tonbo before. According to my experience in some other open source communities, there are two measures. One is to squash when merging. It can be user squash (but it may be for developers), or committer merge. But there is only one commit in the end. I suggest that the commit message is related to the PR title.

**Reply by ethe** [reply]
Sure, I think we could try to use merge / squash merge instead of rebase merge

**Reply by caicancai** [reply]
> Sure, I think we could try to use merge / squash merge instead of rebase merge

yes, If there are no further questions, I will close this discussion.

@ethe Thank you for your patient reply.
